smartos.html

```diff
@@ -195,48 +264,10 @@
 					var ul = document.createElement('ul');
 					ul.className = 'links';
 
-					[
-						{
-							href: BASE_URI + '/' + name + '/smartos-' + name_date + '.iso',
-							name: 'download ISO       smartos-' + name_date + '.iso'
-						},
-						{
-							href: BASE_URI + '/' + name + '/smartos-' + name_date + '-USB.img.' + compressed_suffix,
-							name: 'download USB       smartos-' + name_date + '-USB.img.' + compressed_suffix
-						},
-						{
-							href: BASE_URI + '/' + name + '/smartos-' + name_date + '.vmwarevm.tar.' + compressed_suffix,
-							name: 'download VMware    smartos-' + name_date + '.vmwarevm.tar.' + compressed_suffix
-						},
-						{
-							href: BASE_URI + '/' + name + '/platform-' + name_date + '.tgz',
-							name: 'download Tarball   platform-' + name_date + '.tgz'
-						},
-						null, // spacer
-						{
-							href: BASE_URI + '/' + name + '/index.html',
-							name: 'download page'
-						},
-						{
-							href: BASE_URI + '/' + name + '/changelog.txt',
-							name: 'raw changelog'
-						}
-					].forEach(function (o) {
-						if (o === null) {
-							ul.appendChild(document.createElement('br'));
-							return;
-						}
-						var li = document.createElement('li');
-						var a = document.createElement('a');
-						var pre = document.createElement('pre');
-						pre.style.display = 'inline';
-						a.href = o.href;
-						a.target = '_blank';
-						pre.textContent = o.name;
-						a.appendChild(pre);
-						li.appendChild(a);
-						ul.appendChild(li);
-					});
+					// We determine the platform name by parsing the md5sums
+					// file since these vary by release.
+					var md5_url = BASE_URI + '/' + name + '/md5sums.txt';
+					generate_file_links(md5_url, name, name_date, compressed_suffix, ul);
 
 					// single user root password
 					var rootpwli = document.createElement('li');
```

Comment:
One diff between https://us-east.manta.joyent.com/timf/public/smartos-test.html#20190523T082605Z and https://us-east.manta.joyent.com/Joyent_Dev/public/SmartOS/smartos.html#20190523T082605Z is that in the new one the `- root password "..."` entry is now at the top.  That's a likely unintended side-effect of the additon of the links now being async.

Reply:
Yep, as discussed on mm, moving the root password link generation into the main files-link-generation function we added sorts that out.